client/src/interface/components/ChangePermission/index.js

```diff
@@ -1,192 +1,203 @@
 import React from 'react';
 import Button from '@material-ui/core/Button';
-import Checkbox from '@material-ui/core/Checkbox';
-import FormGroup from '@material-ui/core/FormGroup';
-import FormControlLabel from '@material-ui/core/FormControlLabel';
-import {Box, Dialog, DialogActions, DialogTitle, Divider, Tooltip, Typography} from '@material-ui/core';
-import axios from 'axios';
+import {
+    Box,
+    Checkbox,
+    Dialog,
+    DialogActions,
+    DialogTitle,
+    FormControlLabel,
+    Switch,
+    Table,
+    TableBody,
+    TableCell,
+    TableHead,
+    TableRow,
+    TextField,
+    Tooltip
+} from '@material-ui/core';
 
 import './index.css';
-import * as constants from '../../pages/FileManager/constants';
+import {PERMISSION_BITS} from "../../pages/FileManager/constants";
+import {isDir} from "../../pages/FileManager/utils";
+import {sftp_chmod} from "../../../actions/sftp";
 
 export default class ChangePermission extends React.Component {
     constructor(props) {
         super(props);
+        this.cwd = null
```

Comment:
it seems cwd isn't accessed in the render() func so I moved it out of this.state